File: TestSample/php/src/Speakers/EnSpeaker.php
<?php
declare(strict_types=1);

namespace HelloWorld\Speakers;

use HelloWorld\Person;

final class EnSpeaker extends Person
{
    public function __construct(string $name)
    {
        parent::__construct($name, "en");
    }

    public function speak(): string
    {
        return "Hello, World!";
    }
}

File: TestSample/rust/src/speakers/en_speaker.rs
use crate::person::{Person, PersonBase};

pub struct EnSpeaker {
    base: PersonBase,
}

impl EnSpeaker {
    pub fn new(name: &str) -> Self {
        Self { base: PersonBase::new(name, "en") }
    }
}

impl Person for EnSpeaker {
    fn name(&self) -> &str { &self.base.name }
    fn language(&self) -> &str { &self.base.language }
    fn speak(&self) -> String { String::from("Hello, World!") }
}
